use default esy-installer if esy.install is empty

rauanmayemir opened the pull request:
This PR closes #105.

My assessment was that we shouldn't configure install commands in `esy-build-package`, but do it during the build-plan creation. One of the benefits for this being debugging and having sort of deterministic reviewable build instructions.

Cc: @andreypopp

andreypopp commented:
Looks great! We need to decide what to do with `esy-installer` package (which provides the automatically inserted command of the same name), I see the following options:
1. Check if it is set as a dependency and fail if it's not
2. Distribute `esy-installer` with esy (see #114)

rauanmayemir commented:
I was hoping that since we're planning to move to `esyi`, leave it as it is for now?
So I'd go for checking for `esy-installer` dependency.

jordwalke commented:
`esyi` is of a different nature than `esy-installer`. `esy-installer` doesn't install dependencies, it installs your *built* project into the proper location. It just handles the `.install` file format. `esyi` won't deal with that.

rauanmayemir commented:
I see. Would distributing `esy-installer` with `esy` bloat the size?

jordwalke commented:
Probably a bit. We don't really need to cross that bridge yet because most packages end up including it. As is I think this PR is an improvement. The other thing we can do is make `esy-installer` include a tiny check for the existence of a `.install` file before trying to forward the command to `opam-installer` which it currently does every time it's invoked.

rauanmayemir commented:
Working on resolving the conflicts.

Should we rewrite default `[["esy-installer"]]` to `[["sh", "-c" "(esy-installer || true)"]]`? Some of our shims might fail because they don't build or install anything.

rauanmayemir commented:
Previous diff turned out to be bogus, I fixed it and added tests to confirm it.
Now our own fixtures are tripping up over non-existent `esy-installer` command. 😅

jordwalke commented:
In our shims, should we add a dependency on a dummy esy-installer package that just has a dummy esy-installer executable?

This diff is definitely an improvement to esy configuration.

andreypopp commented:
Sorry for waiting too long before merging this. I think this is an improvement from the UX point of view but was hesitant a bit about the fact that we still require a dependency on `esy-installer` to be specified.
